File: Backend/htbp_parent/service/service_hosp/src/main/java/com/seclib/htbp/hosp/repository/ScheduleRepository.java
package com.seclib.htbp.hosp.repository;

import com.seclib.htbp.model.hosp.Department;
import com.seclib.htbp.model.hosp.Schedule;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

import java.util.Date;
import java.util.List;

@Repository
public interface ScheduleRepository extends MongoRepository<Schedule,String> {
    Schedule getScheduleByHoscodeAndHosScheduleId(String hoscode, String hosScheduleId);

    List<Schedule> findScheduleByHoscodeAndDepcodeAndWorkDate(String hoscode, String depcode, Date toDate);

    Schedule getScheduleById(String scheduleId);
}

File: Backend/htbp_parent/service/service_hosp/src/main/java/com/seclib/htbp/hosp/repository/ScheduleRepository.kt
package com.seclib.htbp.hosp.repository

import org.springframework.data.mongodb.repository.MongoRepository
import com.seclib.htbp.model.hosp.Schedule
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface ScheduleRepository : MongoRepository<Schedule?, String?> {
    fun getScheduleByHoscodeAndHosScheduleId(hoscode: String?, hosScheduleId: String?): Schedule?
    fun findScheduleByHoscodeAndDepcodeAndWorkDate(hoscode: String?, depcode: String?, toDate: Date?): List<Schedule?>?
    fun getScheduleById(scheduleId: String?): Schedule?
}